modules/controllers/event-query-controller.ts

```diff
@@ -126,7 +139,11 @@ export function EventsQueryController(tracer?: any) {
             });
 
             const results = dbEvents.map((event) =>
-                event.type === 'SWAP' ? parseSwap(event as SwapEvent) : parseJoinExit(event as JoinExitEvent),
+                event.type === 'SWAP' && event.protocolVersion === 1
```

Comment:
I'm a bit scared about these protocolVersion tests as they will break if we ever get another custom amm based on v1

Reply:
true, it makes sense to check just for surplus